admin-panel/src/main/java/net/cryptic_game/backend/admin/service/server_management/ServerCommunicationImpl.java

```diff
@@ -0,0 +1,51 @@
+package net.cryptic_game.backend.admin.service.server_management;
+
+import com.nimbusds.jose.shaded.json.JSONObject;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import lombok.RequiredArgsConstructor;
+import net.cryptic_game.backend.admin.Config;
+import net.cryptic_game.backend.admin.exception.InternalServerErrorException;
+import net.cryptic_game.backend.admin.exception.NotFoundException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+import java.nio.charset.StandardCharsets;
+
+@Component
+@RequiredArgsConstructor
+public class ServerCommunicationImpl implements ServerCommunication {
+
+    private final Config config;
+    private HttpClient httpClient;
```

Comment:
Have a loop to the web client from spring. It provides mutch more functionality and could reduce the code.

Reply:
Ps: it is an addition to the reactor Http client.